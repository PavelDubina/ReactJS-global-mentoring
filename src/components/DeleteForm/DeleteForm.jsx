import React from 'react'
import { useDispatch } from 'react-redux'
import PropTypes from 'prop-types'
import { Button } from '../Button/Button'
import { Modal } from '../Modal/Modal'
import deleteMovie from '../../redux/actions/deleteMovie'
import styles from './DeleteForm.scss'

export const DeleteForm = ({ id, onClose }) => {
  const dispatch = useDispatch()
  const onDelete = () => {
    dispatch(deleteMovie(id))
    onClose()
  }
  return (
    <Modal title="DELETE MOVIE" onClose={onClose}>
      <div className={styles.container}>
        <p>Are you sure you want delete this movie?</p>
        <Button onClick={onDelete} styleType="confirm">
          CONFIRM
        </Button>
      </div>
    </Modal>
  )
}
DeleteForm.propTypes = {
  onClose: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
}
