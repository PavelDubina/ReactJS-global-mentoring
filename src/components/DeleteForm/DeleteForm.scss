.container {
  display: flex;
  flex-direction: column;
  padding: 24px 0 0 0;

  & p {
    color: #fff;
    font-size: 18px;
    margin: 0 0 32px 0;
  }
}

button {
  align-self: right;
}
