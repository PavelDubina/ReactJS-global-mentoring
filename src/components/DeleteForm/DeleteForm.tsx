import React from 'react'
import { useDispatch } from 'react-redux'
import { useTranslation } from 'next-i18next'
import { Button } from '../Button/Button'
import { Modal } from '../Modal/Modal'
import deleteMovie from '../../redux/actions/deleteMovie'
import styles from '../../../styles/DeleteForm.module.scss'

type DeleteFormProps = {
  id: number
  onClose: () => void
}

export const DeleteForm: React.FC<DeleteFormProps> = ({ id, onClose }) => {
  const { t } = useTranslation()
  const dispatch = useDispatch()
  const onDelete = () => {
    dispatch(deleteMovie(id))
    onClose()
  }
  return (
    <Modal title={t('deleteModal.title')} onClose={onClose}>
      <div className={styles.container}>
        <p>{t('deleteModal.description')}</p>
        <div className={styles.confirm_btn}>
          <Button onClick={onDelete} styleType="confirm">
            {t('deleteModal.deleteBtn')}
          </Button>
        </div>
      </div>
    </Modal>
  )
}
