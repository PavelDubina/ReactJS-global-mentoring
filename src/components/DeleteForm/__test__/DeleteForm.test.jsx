import React from 'react'
import configureStore from 'redux-mock-store'
import ReactDOM from 'react-dom'
import { render, fireEvent } from '@testing-library/react'
import { Provider } from 'react-redux'
import movies from '../../../MockedData/mockMovies.json'
import { DeleteForm } from '../DeleteForm'

const mockStore = configureStore([])

describe('DeleteForm', () => {
  const store = mockStore({
    moviesData: movies,
  })
  ReactDOM.createPortal = jest.fn((element) => element)
  const getComponent = ({ onClose = jest.fn(), id = 86834 }) =>
    render(
      <Provider store={store}>
        <DeleteForm id={id} onClose={onClose} />
      </Provider>,
    )
  it('Should renders correctly', () => {
    expect(getComponent({}).asFragment()).toMatchSnapshot()
  })
  it('onClose and dispatch have been called', () => {
    store.dispatch = jest.fn()
    const onClose = jest.fn()
    const { getByRole } = getComponent({ onClose })
    fireEvent.click(getByRole('button'))
    expect(store.dispatch).toBeCalled()
    expect(onClose).toBeCalled()
  })
})
