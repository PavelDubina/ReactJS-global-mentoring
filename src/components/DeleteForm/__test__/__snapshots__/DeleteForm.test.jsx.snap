// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteForm Should renders correctly 1`] = `
<DocumentFragment>
  <div
    class="modal_backdrop"
  >
    <div
      class="modal_window"
    >
      <div
        class="close_btn"
      >
        Ã—
      </div>
      <h1
        class="title"
      >
        DELETE MOVIE
      </h1>
      <div
        class="container"
      >
        <p>
          Are you sure you want delete this movie?
        </p>
        <button
          class="confirm"
          type="button"
        >
          CONFIRM
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
